### DISCUSSION TITLE
How to listen to the audio level of webrtc call

### DISCUSSION THREAD

**Discussion by moigamijunior** [pion/webrtc#2428] [Q&A]
How is it possible to listen to the audio level of a WebRTC call with pion?

**Comment by Veercodeprog** [comment]
// Create new instances of AudioContext and RTCPeerConnection
const audioContext = new AudioContext();
const peerConnection = new RTCPeerConnection();

// Request access to the user's microphone and add the resulting audio track to the peer connection
navigator.mediaDevices.getUserMedia({ audio: true }).then(stream => {
  const audioTrack = stream.getAudioTracks()[0];
  peerConnection.addTrack(audioTrack, stream);
});

// Create a new ScriptProcessorNode with a buffer size of 2048
const scriptProcessor = audioContext.createScriptProcessor(2048, 1, 1);

// Connect the local audio track to the ScriptProcessorNode
peerConnection.getSenders()[0].track.connect(scriptProcessor);

// Define a callback function for the onaudioprocess event
scriptProcessor.onaudioprocess = function(event) {
  const inputBuffer = event.inputBuffer.getChannelData(0);
  const audioLevel = Math.abs(inputBuffer.reduce((a, b) => a + b) / inputBuffer.length);
  console.log('Audio level:', audioLevel);
};

// Connect the ScriptProcessorNode to the Audio
